### PR TITLE
kelly/fix_status_check

### PR DESCRIPTION
**Pull request by kellybinary**
- fix check for cashier_locked / withdrawal_locked statuses

### CODE REVIEW

File: src/javascript/app/base/header.js [hunk 1]
```diff
@@ -303,7 +303,7 @@ const Header = (() => {
 
             const validations = {
                 authenticate         : () => +get_account_status.prompt_client_to_authenticate,
-                cashier_locked       : () => /cashier_locked/.test(status),
+                cashier_locked       : () => /^cashier_locked$/.test(status),
```
Comment (negarn) [inline src/javascript/app/base/header.js]: this doesn't work unless the only status in the array is `cashier_locked`. usually we will have multiple items in the status array.

File: src/javascript/app/base/header.js [hunk 2]
```diff
@@ -315,7 +315,7 @@ const Header = (() => {
                 tax                  : () => Client.shouldCompleteTax(),
                 tnc                  : () => Client.shouldAcceptTnc(),
                 unwelcome            : () => /unwelcome/.test(status),
-                withdrawal_locked    : () => /withdrawal_locked/.test(status),
+                withdrawal_locked    : () => /^withdrawal_locked$/.test(status),
```
Comment (negarn) [inline src/javascript/app/base/header.js]: same here.

File: src/javascript/app/base/header.js [hunk 3]
```diff
@@ -299,11 +300,12 @@ const Header = (() => {
                 tnc                  : () => buildMessage('Please [_1]accept the updated Terms and Conditions[_2] to lift your withdrawal and trading limits.',                                  'user/tnc_approvalws'),
                 unwelcome            : () => buildMessage('Trading and deposits have been disabled on your account. Kindly [_1]contact customer support[_2] for assistance.',                    'contact'),
                 withdrawal_locked    : () => localize('Withdrawals have been disabled on your account. Please check your email for more details.'),
+                mt5_withdrawal_locked: () => localize('MT5 withdrawals have been disabled on your account. Please check your email for more details.'),
```
Comment (negarn) [inline src/javascript/app/base/header.js]: this string needs to be added to translations

File: src/javascript/app/base/header.js [hunk 4]
```diff
@@ -299,11 +301,12 @@ const Header = (() => {
...
             };
 
             const validations = {
                 authenticate         : () => +get_account_status.prompt_client_to_authenticate,
-                cashier_locked       : () => /cashier_locked/.test(status),
+                cashier_locked       : () => checkCashierStatus('cashier_locked'),
                 currency             : () => !Client.get('currency'),
                 document_needs_action: () => /document_needs_action/.test(status),
```
Comment (negarn) [inline src/javascript/app/base/header.js]: perhaps we can update all of these to use the same function, and rename it to something more general, like `hasStatus`

Comment (kellybinary) [reply src/javascript/app/base/header.js]: sure, good idea :)

File: src/javascript/app/base/header.js [hunk 5]
```diff
@@ -283,6 +283,7 @@ const Header = (() => {
             };
 
             const buildMessage = (string, path, hash = '') => localize(string, [`<a href="${Url.urlFor(path)}${hash}">`, '</a>']);
+            const hasStatus = (string) => status.findIndex(s => s === string) < 0 ? Boolean(false) : Boolean(true);
```
Comment (negarn) [inline src/javascript/app/base/header.js]: can also be used on line 368

**Pull request merged**